assets/js/modules/analytics-4/components/widgets/PopularProductsWidget.js

```diff
@@ -127,16 +130,23 @@ function PopularProductsWidget( props ) {
 
 	const { rows = [] } = report || {};
 
-	const columns = useSelect( ( select ) => [
+	const columns = [
 		{
 			field: 'dimensionValues',
 			Component: ( { fieldValue } ) => {
 				const [ title, url ] = fieldValue;
-				const serviceURL = select(
-					MODULES_ANALYTICS_4
-				).getServiceReportURL( 'all-pages-and-screens', {
-					filters: { unifiedPagePathScreen: url.value },
-					dates,
+				const serviceURL = useSelect( ( select ) => {
+					return ! viewOnlyDashboard
+						? select( MODULES_ANALYTICS_4 ).getServiceReportURL(
+								'all-pages-and-screens',
+								{
+									filters: {
+										unifiedPagePathScreen: url.value,
+									},
+									dates,
+								}
+						  )
+						: null;
 				} );
```

Comment:
This was actually intentional and explained to me by @hussain-t in Slack… because this `useSelect` is _for_ this nested component, it needed to be relocated. Makes sense!

This pattern is apparently being used in a few other spots too, but we should maybe add a comment explaining it, just because it isn't immediately obvious what's happening here and it might catch someone off-guard 🙂

Reply:
Added a code comment ✅